# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/FramebufferImpl.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/FramebufferImpl.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/FramebufferImpl.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/FramebufferImpl.h	Fri Nov 08 17:06:00 2019 +0000
@@ -86,11 +86,23 @@
                                             size_t index,
                                             GLfloat *xy) const = 0;
 
+    // Special configuration option for checkStatus(). Some back-ends don't require a syncState
+    // before calling checkStatus. In practice the GL back-end is the only config that needs
+    // syncState because it depends on the behaviour of the driver. Allowing the Vulkan and
+    // D3D back-ends to skip syncState lets us do more work in the syncState call.
+    virtual bool shouldSyncStateBeforeCheckStatus() const;
+
     const gl::FramebufferState &getState() const { return mState; }
 
   protected:
     const gl::FramebufferState &mState;
 };
+
+inline bool FramebufferImpl::shouldSyncStateBeforeCheckStatus() const
+{
+    return false;
+}
+
 }  // namespace rx
 
 #endif  // LIBANGLE_RENDERER_FRAMEBUFFERIMPL_H_